tests/pos/param-depmeth.scala

```diff
@@ -0,0 +1,15 @@
+object Test {
+
+  class C { type T }
+
+  def f(x: C, y: x.T): x.T = y // ok
+
+  val c = new C { type T = String }
+  val c2 = c
```

Comment:
I think we only added `val c2` to see what type is inferred, so we can now remove it, no? A similar thing goes for the `val d` below, I think.